python/ray/data/_internal/logical/operators/from_blocks_operator.py

```diff
@@ -0,0 +1,67 @@
+import functools
+from typing import List, Optional
+
+from ray.data._internal.execution.interfaces import RefBundle
+from ray.data._internal.logical.interfaces import LogicalOperator
+from ray.data._internal.util import unify_block_metadata_schema
+from ray.data.block import Block, BlockMetadata
+
+
+class FromBlocks(LogicalOperator):
+    """Logical operator for in-heap-memory (not object store) input data.
+
+    If you want to create a Dataset from in-object-store data, use the `InputData`
+    logical operator instead.
+    """
+
+    def __init__(
+        self,
+        input_blocks: List[Block],
+        input_metadata: List[BlockMetadata],
+    ):
+        super().__init__(self.__class__.__name__, [], len(input_blocks))
+        assert len(input_blocks) == len(input_metadata), (
```

Comment:
Let's just pass a list of tuples

Reply:
Will address in follow-up since this PR maintains the current interface